libnetdata/socket/security.c

```diff
@@ -127,7 +128,11 @@ void security_start_ssl(int type) {
         netdata_srv_ctx =  security_initialize_openssl_server();
     }
     else {
-        netdata_cli_ctx =  security_initialize_openssl_client();
+        if(type == 1) {
+            netdata_cli_ctx = security_initialize_openssl_client();
+        } else if (type == 2) {
```

Comment:
Magic number?

Reply:
`switch` would be easier to read than a bunch of `if...else`

Reply:
It is not a magic number exactly, it was an order defined during  the development, firstly we developed the server(0), the next was the stream(1) and finally we began to work with backends, and the first backend is opentsdb(2);

For this specific case that we have only 3, I agree that tje switch can be used, but this is not always the case.

Reply:
This does look like a magic number indeed if i may say

Reply:
I meant that it would be better to use macros instead of 0, 1, 2.

Reply:
I think the following code is easier to read:
```c
void security_start_ssl(int type) {
    switch (type) {
        case NETDATA_SSL_CONTEXT_SERVER:
            {
                struct stat statbuf;

                if (stat(security_key, &statbuf) || stat(security_cert, &statbuf)) {
                    info("To use encryption it is necessary to set \"ssl certificate\" and \"ssl key\" in [web] !\n");
                    return;
                }
            }
            netdata_srv_ctx =  security_initialize_openssl_server();
            break;
        case NETDATA_SSL_CONTEXT_STREAMING:
            netdata_cli_ctx = security_initialize_openssl_client();
            break;
        case NETDATA_SSL_CONTEXT_OPENTSDB:
            netdata_opentsdb_ctx = security_initialize_openssl_client();
            break;
    }
}
```

Reply:
After the talk on Friday night, I was convinced that I did not need to avoid more the "switch", definitively the compilers improved a lot since the first time I executed a reverse engineer and  I compared both "if ... else if " with "swtich".